geps/gep-1709.md

```diff
@@ -37,8 +45,304 @@ which implementations can prove they satisfy and be recognized for.
+
+Implementations will be able to report their conformance testing results using
+our [reporting process](#reporting-process). Implementations will be able to
+visibly demonstrate their conformance results on their downstream projects and
+repositories using our [certification process](#certification-process).
+
+#### Reporting Process
+
+When conformance tests complete a log message will be emitted:
+
+```console
+Success! Conformance profiles "Layer7" and "Layer4" passed for version v0.7.0.
+The following extended features were flagged as not supported for "Layer7Extended":
+- HTTPResponseHeaderModification
+- HTTPRouteQueryParamMatching
+```
+
+For the above output upstream Gateway API can manually generate or update a report
+in the following format:
+
+```json
+apiVersion: v1alpha1
+kind: ConformanceReport
+implementation: acmeorg-acme
+reports:
+  layer4:
+    releases:
+    - tag: v0.7.0
+    - tag: v0.6.2
+    - tag: v0.6.1
+  layer7:
+    releases:
+    - tag: v0.7.0
+      extended:
+        unsupportedFeatures:
+        - HTTPResponseHeaderModification
+        - HTTPRouteQueryParamMatching
+    - tag: v0.6.2
+      extended:
+        unsupportedFeatures:
+        - HTTPResponseHeaderModification
+        - HTTPRouteQueryParamMatching
+        - HTTPRouteMethodMatching
+    - tag: v0.6.1
+      extended:
+        unsupportedFeatures:
+        - HTTPResponseHeaderModification
+        - HTTPRouteQueryParamMatching
+        - HTTPRouteMethodMatching
+maintainers:
+- @kubernetes-sigs/gateway-api-maintainers
+```
+
+> **NOTE**: In the above the `implementation` field is a combination of
+> `<organization>-<project>`. Organizations can be an open source organization,
+> an individual, a company, e.t.c.. Organizations can theoretically have more
+> than one `<project>` name and submit separate reports for each of them.
+
+> **NOTE**: The above demonstrates that the `acmeorg-acme` project was already
+> passing for versions prior to `v0.7.0`. This is how we will track history of
+> conformance for releases across implementations, but at some point might
+> require some archival process (which we probably don't have to worry about
+> for this iteration). You can also see that there's been progress to add the
+> `HTTPRouteMethodMatching` for the latest release.
+
+> **NOTE**: The `maintainers` field indicates the Github usernames or team
+> names of those who are responsible for maintaining this file, so they can be
+> easily contacted when needed (e.g. for relevant release announcements
+> regarding conformance, e.t.c.).
+
+The implementation can compile its `ConformanceReport` and upload it to the
+Gateway API project by creating a pull request. The file should be uploaded
+to the following sub-directory:
+
+```console
+conformance/results/<organization>-<project>.yaml
+```
+
+> **NOTE**: For this iteration we're focusing on getting these reports created
+> but at first they will probably be created by hand. In future iterations and
```

Comment:
might want to add somewhere that we should have CI in this repo to do some YAML linting and/or schema validation of these uploaded reports

implementation detail but maybe good to point out as a requirement if we're starting to do this with "hand crafted" YAML

Reply:
sg, added a TODO for this in cfcc08351432dfe239ff18dd295ff357f55716ef, we will need to resolve this and other TODO's prior to going `implementable` :+1: